File: solutions/0350_Intersection_of_Two_Arrays_II/hashmap.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        Map<Integer, Integer> counter = new HashMap<>();
        for (int v : nums1) {
            counter.put(v, counter.getOrDefault(v, 0) + 1);
        }
        List<Integer> ret = new ArrayList<>();
        for (int v : nums2) {
            if (counter.containsKey(v) && counter.get(v) > 0) {
                ret.add(v);
                counter.put(v, counter.get(v) - 1);
            }
        }
        return ret.stream().mapToInt(i->i).toArray();
    }
}

File: solutions/0387_First_Unique_Character_in_a_String/hashmap.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        counter = {}
        for i, c in enumerate(s):
            if c in counter:
                counter[c] = 1 << 31
            else:
                counter[c] = i
        ret = min(counter.values())
        return -1 if ret == 1 << 31 else ret
